src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -139,19 +140,36 @@ export class TerminalInstance implements ITerminalInstance {
 		this._initDimensions();
 		this._createProcess(this._shellLaunchConfig);
 		this._createXterm();
+		if (this._shellLaunchConfig) {
+			this.setTitle(path.basename(this._shellLaunchConfig.executable), true);
```

Comment:
If `this._shellLaunchConfig.name` is set, we do not want to do this.

Also this applies to Linux/macOS too, I don't think we use `basename` for non-Windows yet. This also doesn't strip .exe, we should rely on the `basename`/.exe strip code inside `setTitle`.